### PR TITLE
Add GDScript support

### PR DESCRIPTION
**Pull request by Shatur95**
Recently Godot 3.2 was released with LSP support. But LSP works only through Websocket. [This](https://github.com/godotengine/godot/pull/35864#) PR switches Godot's LSP to TCP and now can be supported for the Vim plugins. This PR adds support for upcoming Godot version via `nc` (works only with mentioned PR above).

### CODE REVIEW

File: lua/nvim_lsp/gdscript.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+configs.gdscript = {
+  default_config = {
+    cmd = {"nc", "localhost", "6008"};
+    filetypes = {"gd", "gdscript3"};
+    root_dir = util.root_pattern("project.godot", ".git");
+    log_level = lsp.protocol.MessageType.Warning;
+  };
+  -- on_new_config = function(new_config) end;
+  -- on_attach = function(client, bufnr) end;
```
Comment (h-michael) [inline lua/nvim_lsp/gdscript.lua]: We don't need these lines.

File: lua/nvim_lsp/gdscript.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  docs = {
+    description = [[
+https://github.com/godotengine/godot
+
+Language server for GDScript, used by Godot Engine.
+]];
+    default_config = {
+      root_dir = "vim's starting directory";
```
Comment (h-michael) [inline lua/nvim_lsp/gdscript.lua]:
```suggestion
      root_dir = util.root_pattern("project.godot", ".git");
```

File: lua/nvim_lsp/gdscript.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    root_dir = util.root_pattern("project.godot", ".git");
+    log_level = lsp.protocol.MessageType.Warning;
+  };
+  -- on_new_config = function(new_config) end;
+  -- on_attach = function(client, bufnr) end;
+  docs = {
+    description = [[
+https://github.com/godotengine/godot
+
+Language server for GDScript, used by Godot Engine.
```
Comment (h-michael) [inline lua/nvim_lsp/gdscript.lua]: Could you describe a link to the LSP setup method or write a method?

Comment (Shatur95) [reply lua/nvim_lsp/gdscript.lua]: Unfortunately, godot does not currently have LSP documentation. I learned about it from the corresponding [extension](https://github.com/godotengine/godot-vscode-plugin) for VSCode.

**Comment by Shatur95** [comment]
> Would you give me some simple steps to try this?

Just install Godot Engine and open any project from the template library. Than open any project *.gd file in Neovim.

**Comment by h-michael** [comment]
Would you tell me a samle project?

**Comment by Shatur95** [comment]
I meant that you can just open any sample project in projects library (integrated in editor).
But you can also try [this one](https://github.com/Zoriya/zoriya) :)

**Comment by h-michael** [comment]
Thanks, I:ll try it latter :)

**Comment by h-michael** [comment]
Sorry to late

**Pull request merged**